src/map/skill.cpp

```diff
@@ -14967,6 +14965,9 @@ bool skill_check_condition_castbegin(struct map_session_data* sd, uint16 skill_i
 				return false;
 			}
 			break;
+		case AL_HOLYWATER:
+			if(pc_search_inventory(sd,ITEMID_EMPTY_BOTTLE) < 0)
```

Comment:
I'm just thinking if it makes sense to put this into skill_require_db. But I cannot remember anymore where we had a similar requirement. I think it was one of the Mechanic skills... Just look for something like this code snippet:
`struct skill_condition require = skill_get_requirement(sd, skill_id, skill_lv);`

Reply:
Also it looks weird without a break in this switch case.

Reply:
But adding it to the skill_require_db would make it return a "skill failed" message right?
Btw when testing this pc_search_inventory, I noticed it returns -1 when have no empty bottles, and 15 always when have 1 or more, weird.
It doesn't need a break, since it just checks if you don't have bottles does nothing, otherwise continues to the ST_WATER case.

Reply:
It returns the inventory index(-1 if not found, 0 to MAX_INVENTORY=100 if found).
You can then access the player's inventory with this index and check the count.

I'm just wondering because the case was not there and I'm a bit unsure if the skill level of HOLYWATER should really be increased by SC_RAISINGDRAGON.
The water check should be done elsewhere I guess, since the HOLYWATER case has not been there before and it was still checked if you are standing on water.

Reply:
No, that belongs to the skill MO_CALLSPIRITS and isn't related to my changes XD
The case I added just forces to do nothing when the player doesn't have any empty bottles.
Without this, the skill would cast and show a skill failed message, and that's not what happens in Aegis, it should do just nothing.

Reply:
Since you did not add a break it will run through these cases as well. That's what I'm saying!
See:
"Also it looks weird without a break in this switch case."

Reply:
Ohh, right!

Reply:
# is used for issues and pull requests :)
But now it's good! 👍